presto-bytecode/src/main/java/com/facebook/presto/bytecode/MethodDefinition.java

```diff
@@ -236,19 +235,19 @@ public void visit(ClassVisitor visitor, boolean addReturn)
                 parameterAnnotation.visitParameterAnnotation(parameterIndex, methodVisitor);
             }
         }
-
-        // visit code
-        methodVisitor.visitCode();
-
-        // visit instructions
-        MethodGenerationContext generationContext = new MethodGenerationContext(methodVisitor);
-        generationContext.enterScope(scope);
-        body.accept(methodVisitor, generationContext);
-        if (addReturn) {
-            new InsnNode(RETURN).accept(methodVisitor);
+        if (!declaringClass.isInterface()) {
+            // visit code
+            methodVisitor.visitCode();
+
+            // visit instructions
+            MethodGenerationContext generationContext = new MethodGenerationContext(methodVisitor);
+            generationContext.enterScope(scope);
+            body.accept(methodVisitor, generationContext);
```

Comment:
verify body is empty for interfaces... or even better throw an exception from the `MethodDeclaration.getBody` if the class is an interface